src/tls/openssl.c

```diff
@@ -529,9 +529,15 @@ static int tls_net_handshake(struct flb_tls *tls,
             pthread_mutex_unlock(&ctx->mutex);
             return -1;
         }
-
-        if (vhost != NULL) {
-            SSL_set_tlsext_host_name(session->ssl, vhost);
+        if (tls->vhost != NULL) {
+            SSL_set_tlsext_host_name(session->ssl, tls->vhost);
+            /* set host name validation only if vhost is configured
+             * explicitely */
+            X509_VERIFY_PARAM *param = SSL_get0_param(session->ssl);
+            if (!X509_VERIFY_PARAM_set1_host(param, tls->vhost, 0)) {
```

Comment:
Move the call to `X509_VERIFY_PARAM_set1_host` out of the `if` condition, save its return value in `ret` and then compare `ret` in the `if ` condition to `0` explicitly so it's  clear that it's not treated as an implicit boolean value.